xterm/index.cgi

```diff
@@ -179,7 +179,10 @@ if (!-r $shellserver_cmd) {
 	}
 my $tmpdir = &tempname_dir();
 $ENV{'SESSION_ID'} = $main::session_id;
-&system_logged($shellserver_cmd." ".quotemeta($port)." ".quotemeta($user).
+&foreign_require('acl');
+my $parentsessionid = &acl::hash_session_id($main::session_id);
```

Comment:
Passing in the session ID as a command-line parameter is a bad idea, because it will be visible to anyone on the system running `ps` and could be used to hijack another user's session!  Also, it's available via the `$main::session_id` variable anyway.

Reply:
I wasn't passing session id as is, as the session id was hashed.